maven-model-builder/src/main/java/org/apache/maven/model/building/DefaultModelBuilder.java

```diff
@@ -482,6 +500,68 @@ else if ( !parentIds.add( parentData.getId() ) )
         return resultModel;
     }
 
+    private Map<String, Activation> getInterpolatedActivations( Model rawModel,
+                                                                DefaultProfileActivationContext context,
+                                                                DefaultModelProblemCollector problems )
+    {
+        Map<String, Activation> interpolatedActivations = getProfileActivations( rawModel, true );
+        for ( Activation activation : interpolatedActivations.values() )
+        {
+            if ( activation.getFile() != null )
+            {
+                replaceWithInterpolatedValue( activation.getFile(), context, problems );
+            }
+        }
+        return interpolatedActivations;
+    }
+
+    private void replaceWithInterpolatedValue( ActivationFile activationFile, ProfileActivationContext context,
+                                               DefaultModelProblemCollector problems  )
+    {
+        String path;
+        boolean missing;
+
+        if ( isNotEmpty( activationFile.getExists() ) )
+        {
+            path = activationFile.getExists();
+            missing = false;
+        }
+        else if ( isNotEmpty( activationFile.getMissing() ) )
+        {
+            path = activationFile.getMissing();
+            missing = true;
+        }
+        else
+        {
+            return;
+        }
+
+        try
```

Comment:
I think moving the try block to surround everything, and combining the if-else blocks inside it should work.

Reply:
answered above.